src/WordPress/Hooks.php

```diff
@@ -478,6 +479,31 @@ public function purgeCacheOnNewComment($comment_id, $comment_status, $comment_da
         $this->purgeCacheByRelevantURLs($comment_data['comment_post_ID']);
     }
 
+    public function purgeCacheOnTermChange($ids, $taxonomy, $clean_taxonomy)
+    {
+        $args = array(
+            'posts_per_page' => -1,
```

Comment:
This is going to introduce performance issues on large sites as it's going to query for _all_ posts.

It feels like this action itself should be optional and limited to the first ~100 or so posts?

Reply:
within a certain taxonomy yes.  In is not the most efficient way to query but I can't think of another way to determine associated posts.

I get your thought about top 100 but what is worse, slower execution for a site author, or incorrect pages for sure visitors.

In another pull request, I've somewhat addressed the site author performance by offloading the related URL purge to cron